streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -180,4 +205,21 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
   private def extractDistribution(getMetric: BatchInfo => Option[Long]): Option[Distribution] = {
     Distribution(completedBatchInfos.flatMap(getMetric(_)).map(_.toDouble))
   }
+
+  def getBatchInfo(batchTime: Time): Option[BatchInfo] = synchronized {
+    waitingBatchInfos.get(batchTime).orElse {
+      runningBatchInfos.get(batchTime).orElse {
+        completedBatchInfos.find(batch => batch.batchTime == batchTime)
+      }
+    }
+  }
+
+  def getJobInfos(batchTime: Time): Option[Seq[(OutputOpId, JobId)]] = synchronized {
+    batchTimeToJobIds.get(batchTime).map(_.toList)
+  }
+}
+
+private[streaming] object StreamingJobProgressListener {
+  type JobId = Int
```

Comment:
As I said at other places, `JobId` is confusing its not clear whether it is a streaming job or a spark job. Please call it `SparkJobId`. And also please rename all the places you are passing spark's job id as `sparkJobId`.